src/libtexture/imagecache.cpp

```diff
@@ -3438,18 +3478,154 @@ namespace {
 
 
 
+void
+ImageCacheFile::udim_setup()
+{
+    // If it's a literal, existing file, always treat it as a regular
+    // texture, even if it has what looks like udim pattern markers.
+    if (Filesystem::exists(m_filename)) {
+        m_is_udim = false;
+        return;
+    }
+
+    // Look for udim pattern markers, which identify this as a udim pattern.
+    // We separately identify whether it's the kind of pattern that has a
+    // single index, or separate u and v indices (and for separate, whether
+    // we use the convention of adding 1 to the tile index) -- we're going
+    // to need to know that later.
+    bool udim1   = (Strutil::contains(m_filename, "<UDIM>")
```

Comment:
I mean, this is splitting hairs, but it's important to note that the patterns are always just before the file extension (almost in every case).
For for a little more more performance, ensure you reverse search for the pattern, or reverse search from the dot before the file extension.
But we're talking tiny savings overall here.

Reply:
I don't feel comfortable making too any assumptions about how people will structure the `<UDIM>` marker within their filenames, that sounds inherently risky. At our place, I think the UDIM designation might be before the color space and the frame number if there is one (because why not, you could have an animated texture even in a udim context).

This section of code only happens once per UDIM pattern for the whole render, I don't see how it could possibly be a performance issue.

Reply:
Maybe it's easier to jist say that a reverse search here is likely to be faster than a forward search. I.e. rfind vs find. Yes it's not performance critical code, but every wasted cycle is another dead tree.

Reply:
OH, I see what you're saying now. That's an interesting observation! I'll tackle that in the next round as well.